server/schedule/operator/builder.go

```diff
@@ -676,35 +668,49 @@ func (b *Builder) execRemovePeer(peer *metapb.Peer) {
 }
 
 func (b *Builder) execChangePeerV2(needEnter bool, needTransferLeader bool) {
-	// Enter
-	step := ChangePeerV2Enter{
-		PromoteLearners: make([]PromoteLearner, 0, len(b.toPromote)),
-		DemoteVoters:    make([]DemoteVoter, 0, len(b.toDemote)),
-	}
+	if len(b.toPromote)+len(b.toDemote) == 0 {
+		// No need to add empty enter / leave joint consensus step if no peer in toPromote and toDemote
 
-	for _, p := range b.toPromote.IDs() {
-		peer := b.toPromote[p]
-		step.PromoteLearners = append(step.PromoteLearners, PromoteLearner{ToStore: peer.GetStoreId(), PeerID: peer.GetId()})
-		b.currentPeers.Set(peer)
-	}
-	b.toPromote = newPeersMap()
+		// Transfer Leader
+		if needTransferLeader && b.originLeaderStoreID != b.targetLeaderStoreID {
+			b.execTransferLeader(b.targetLeaderStoreID)
+		}
+	} else {
```

Comment:
```suggestion
		}
		return
	}
```

Reply:
The content in the `else` is too long. In fact, `len(b.toPromote)+len(b.toDemote) == 0` is only a special case, so you can `return` directly in the `if`, and no `else` is needed later.